### ISSUE TITLE
Invalid expression: SyntasticStatuslineFlag() keeps Syntastic from loading

### ISSUE THREAD

**Issue by jaredvacanti** [scrooloose/syntastic#1425]
I also have an SO issue here, can't fix the issue on IRC or here.  
http://stackoverflow.com/questions/30387413/syntastic-error-on-load-with-vundle-macvim

**.vimrc**

    syntax enable colorscheme monokai
    set nocompatible
    filetype off
    filetype plugin indent on
    set hidden
    syntax on
    set wildmenu
    set showcmd
    set ignorecase
    set smartcase
    set nostartofline
    set ruler
    set laststatus=2
    set confirm
    set visualbell
    set mouse=a
    set number
    set tabstop=4
    set shiftwidth=4
    set expandtab

**.gvimrc**

    set lines=60 columns=100
    colorscheme monokai
    set number
    set nocompatible
    filetype off
    highlight Pmenu ctermfg=2 ctermbg=3 guifg=#000000 guibg=#ededed
    
    set rtp+=~/.vim/bundle/vundle
    call vundle#rc()
    
    Bundle 'gmarik/Vundle.vim'
    Bundle 'Valloric/YouCompleteMe'
    Bundle 'scrooloose/syntastic'
    Bundle 'bling/vim-airline'
    Bundle 'scrooloose/nerdcommenter'
    Bundle 'tpope/vim-fugitive'
    Bundle 'tpope/vim-surround'
    
    call vundle#end()
    
    filetype plugin indent on    " required
    
    
    "set statusline+=%#warningmsg#
    "set statusline+=%{SyntasticStatuslineFlag()}
    "set statusline+=%*
    
    set statusline+=%#warningmsg#
    set statusline+=%{SyntasticStatuslineFlag()}
    set statusline+=%*
    
    let g:syntastic_always_populate_loc_list = 1
    let g:syntastic_auto_loc_list = 1
    let g:syntastic_check_on_open = 1
    let g:syntastic_check_on_wq = 0

    "let g:statline_syntastic = 0
    "let g:syntastic_check_on_open=1
    "let g:syntastic_enable_signs=1
    "let g:syntastic_always_populate_loc_list = 1
    "let g:syntastic_auto_loc_list = 1
    "let g:syntastic_check_on_open = 1
    "let g:syntastic_check_on_wq = 0
    "let g:syntastic_html_tidy_ignore_errors = [
    "     \   '<img> escaping malformed URI reference'
    "     \ ]
    "set statusline+=%#warningmsg#
    "set statusline+=%{SyntasticStatuslineFlag()}
    "set statusline+=%*
    
    autocmd FileType python nnoremap <buffer> <F5> :exec '!python' shellescape(@%, 1)<cr>
    
    set backupdir=./.backup,.,/tmp
    set directory=.,./.backup,/tmp 

I'm getting the error 
>Unknown function: SyntasticStatuslineFlag
>Invalid expression: SyntasticStatuslineFlag() 

There was a previous closed issue saying to add the line `let g:statline_syntastic = 0` but that the newer version fixed the issue. It doesn't work with or without the line for me. I've updated to the current version of syntastic, using Custom Version 7.4 (71) of MacVim (https://github.com/joelcogen/homebrew-macvimsplitbrowser) on Mac OSX 10.10.3 (14D136). Is this an issue with syntastic? I've reproduced the issue on other machines using the same .vimrc and .gvimrc.

**Comment by lcd047** [comment]
I took the liberty to edit your message to copy your config from your SO post.

You have an unusual setup here.  Please move everything from your `gvimrc` to the end of your `gvim`, except the lines
```vim
set lines=60 columns=100
colorscheme monokai
highlight Pmenu ctermfg=2 ctermbg=3 guifg=#000000 guibg=#ededed
```
Also `autocmd FileType` doesn't make sense after `filetype off`.

After you fix that, please update all your plugins, disable YouCompleteMe, and see if the problem still persists.

**Comment by lcd047** [comment]
Ping.

**Comment by jaredvacanti** [comment]
The problem was still there. I hate to say my solution was a workaround -- I rebuilt MacVim from source (https://github.com/macvim-dev/macvim), used the newer source of Vim that is installed with MacVim executable as an alias, and reinstalled my Vundle bundles. This way, I can have the lines you mentioned put into my vimrc without giving trouble (whereas before my system Vim was not updated enough, prebuilt darwin 7.3, especially with YCM).

To fill the file browser goal I just switched to NERDTree. Not much closure for this issue, but those steps did fix my immediate problem. I'll close now unless opposed.
